fakebook/src/app/model/notification.ts

```diff
@@ -0,0 +1,8 @@
+export interface Notification {
+    // covers all cases of notifications, postId will be left undefined
+    // in the case of a follow, where userId is the person following/unfollowing
+    // the four cases covered by type are: "Like", "Comment", "Post", "Follow"
+    userId: number;
+    postId: number | undefined;
+    type: string;
```

Comment:
make type have type `'Like' | 'Comment' | 'Post' | 'Follow'` then.
in fact, use `type` instead of `interface` here, define four types for each kind of notification, and define `type Notification = LikeNotification | CommentNotification | PostNotification | FollowNotification`